### ISSUE TITLE
social signup form crashes on email collision (trying to generate 'email_taken' error message)

### ISSUE THREAD

**Issue by panbolec** [pennersr/django-allauth#3175]
1. I have a working account, registerd as normal account with given email
2. Then I try to sign in with social account with the same email address (tested with google and facebook)
3. So, I get redirected to social signup form (since ACCOUNT_UNIQUE_EMAIL is True)
4. At "/accounts/social/signup/" I see email field filled in the form and name field empty. 

Whatever I fill in name field and submit the form, the site crashes with 500. 
It tries to generate email_taken error message but there is an exception in 
socialaccount\forms.py in validate_unique_email in

    raise forms.ValidationError(
          get_adapter().error_messages["email_taken"]
          % self.sociallogin.account.get_provider().name
     )

"TypeError: not all arguments converted during string formatting"
because self.sociallogin.account does not exists. 

(allauth.socialaccount.models.SocialAccount.user.RelatedObjectDoesNotExist: SocialAccount has no user)

**Comment by derthadams** [comment]
I'm also getting a 500 error in a similar scenario, and I think I've found the cause. The issue seems to have happened between versions 0.51 and 0.52. 

If a user tries to sign up with a social account whose email already exists in the database, they're redirected to `accounts/social/signup`, where the email field is pre-filled with the pre-existing email, and the submit button says "Sign Up".

In version 0.51, selecting "Sign Up" without changing the pre-existing email causes the page to reload with the error message in `SocialAccountAdapter.error_messages["email_taken"]` appearing above the email field. Continuing to select "Sign Up" with the same email just brings the user back to the same page with the same error message.

In version 0.52, selecting "Sign Up" without changing the pre-existing email causes  a `django.db.utils.IntegrityError`, which in production leads to a 500 response. The new account is being created despite the duplicate email, which is causing a database integrity error.

I've narrowed it down to changes in the `clean_email` method of `account.forms.BaseSignupForm` in commit [a807e5d](https://github.com/pennersr/django-allauth/commit/a807e5db411c4762310552bb4b882ca9285888f1#diff-aca468f72aa7648b36b8ee86a25c3ade350120bc2769cd875e418b1f50574935L343-L348), "Prevent enumeration at signup":

```python
def clean_email(self):
        value = self.cleaned_data["email"]
        value = get_adapter().clean_email(value)
        if value and app_settings.UNIQUE_EMAIL:
            try:
                value = self.validate_unique_email(value)
            except forms.ValidationError:
                if not self.prevent_enumeration:
                    raise
                self.account_already_exists = True
        return value
```

Previously, the try/except block was missing:

```python
def clean_email(self):
        value = self.cleaned_data["email"]
        value = get_adapter().clean_email(value)
        if value and app_settings.UNIQUE_EMAIL:
            value = self.validate_unique_email(value)
        return value
```

With the changes, if the email is a duplicate and `prevent_enumeration` is set to `True`, the `forms.ValidationError` is suppressed, so that the user is not taken back to `accounts/social/signup` (and the information that an account with the email already exists is not revealed to a potential attacker). 

Instead, the `account_already_exists` property of the `BaseSignupForm` is set to `True`. This property is [used in `account.SignupForm.save()`](https://github.com/pennersr/django-allauth/blob/master/allauth/account/forms.py#L438-L442) to prevent the new account from being created (an email is sent instead informing the user that they already have an account).

However, in `socialaccount.SignupForm.save()`, there is no corresponding check of `account_already_exists`, so the new account with the duplicate email is created, leading to the database integrity error.
